apps/system/js/browser_context_menu.js

```diff
@@ -51,6 +50,8 @@
         return;
       }
 
+      SettingsListener.observe(PINNING_PREF, '', this._setPinning.bind(this));
+
```

Comment:
If you follow this comment, you may also want to move the code that remove the observer after the:
     if (!this.contextMenuView.isShown()) {
        return;
      } 
code in the hide method of BrowserContextMenu.

All that said I wonder if this may not generate a race if the settings takes some time to come back from the mozSettings database. In this case you may want to keep the initial observe code in the constructor as it was before this code. And the call to this.hideContextMenu you have added in appWindow will take care of cleaning it (I think it will be better to have a more explicit destroy() method in contextmenu that calls both hide and remove the observer so you will not add/remove the observer all the time in show/hide and explicitely kill the contextmenu to let it do its own cleanup).